flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/operator/common/statistics/basicstatistic/TableSummarizer.java

```diff
@@ -0,0 +1,404 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.operator.common.statistics.basicstatistic;
+
+import org.apache.flink.ml.common.linalg.DenseMatrix;
+import org.apache.flink.ml.common.linalg.DenseVector;
+import org.apache.flink.ml.common.linalg.MatVecOp;
+import org.apache.flink.ml.common.linalg.VectorUtil;
+import org.apache.flink.types.Row;
+
+import java.util.Arrays;
+
+/**
+ * It is summarizer for table， it will calculate statistics and return TableSummary.
+ * You can get statistics from Summary.
+ */
+public class TableSummarizer extends BaseSummarizer {
```

Comment:
Can we factor out Table part into a separated PR? seems like they should belong to [FLINK-13923](https://issues.apache.org/jira/browse/FLINK-13923)